src/backends/pytorch/libtorch_backend.cc

```diff
@@ -394,21 +433,52 @@ Status
 LibTorchBackend::Context::ReadFixedSizedOutputTensor(
     std::vector<torch::Tensor>* outputs_, const std::string& name,
     const int& op_index, const DataType dtype, const size_t dtype_byte_size,
-    const size_t total_batch_size, std::vector<Scheduler::Payload>* payloads)
+    const size_t total_batch_size, std::vector<Scheduler::Payload>* payloads,
+    const DimsList& dims)
 {
   std::vector<int64_t> content_shape;
   void* content = nullptr;
   size_t byte_size = 0;
   RETURN_IF_ERROR(GetOutputTensor(
-      outputs_, op_index, dtype, &content, &byte_size, &content_shape));
+      outputs_, op_index, name, dtype, &content, &byte_size, &content_shape));
+
+  // verify shape of output matches shape from model config
+  const int batch_offset = ((max_batch_size_ == NO_BATCHING) ? 0 : 1);
+
+  for (int i = 0; i < dims.size(); i++) {
+    if (dims[i] != -1) {
+      if (dims[i] != content_shape[i + batch_offset]) {
+
+        // create print friendly array of shapes
```

Comment:
See src/core/model_config.h DimsListToString... you don't need to write your own versions.

Reply:
Got it